### PR TITLE
Implement !help command

### PR DESCRIPTION
**Pull request by raccube**

### CODE REVIEW

**Comment by raccube** [comment]
Example Screenshot:
screenshot 2018-10-29 at 9 21 10 pm

File: src/commands/help.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  static commandDescription() {
+    return 'Shows this message'
+  }
+
+  command() {
+    // const messageArguments = this.message.content
+    //   .slice(this.config.get('prefix').length)
+    //   .split(/ +/)
```
Comment (pixeldesu) [inline src/commands/help.ts]: If you don't use it, remove it!

File: src/commands/help.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    const commandManager = new CommandManager(this.config, database)
+    const commands = commandManager.collectCommands()['message']
+    const message = new RichEmbed()
+    for (const command in commands) {
+      message.addField(command, commands[command].commandDescription())
```
Comment (pixeldesu) [inline src/commands/help.ts:32]: Add an enabled check to the command field loop!

```suggestion
      import isCommandEnabled from '../utilities/isCommandEnabled'
      // ...
      isCommandEnabled(command).then(result => {
      	if (result) {
            message.addField(command, commands[command].commandDescription())	
      	}
      }
```

**Review by pixeldesu — CHANGES_REQUESTED** [review]
Looks good to me so far, only 2 small things I have to address/suggest, otherwise fine!

File: src/commands/help.ts (hunk 2 above)
Comment (raccube) [reply src/commands/help.ts:32]: This won't work as the promise won't resolve before the message gets sent therefore the message will not contain any fields.

**Review by pixeldesu — APPROVED** [review]

**Pull request merged**